cmd/gateway-common.go

```diff
@@ -331,6 +331,8 @@ func ErrorRespToObjectError(err error, params ...string) error {
 		err = InvalidUploadID{}
 	case "EntityTooSmall":
 		err = PartTooSmall{}
+	case "MethodNotAllowed":
+		err = MethodNotAllowed{}
```

Comment:
Can you not add this as well to avoid `minio-go` dependency on this change? 

```diff
git diff
diff --git a/cmd/gateway-common.go b/cmd/gateway-common.go
index 1c74c29ff..c39ceb54c 100644
--- a/cmd/gateway-common.go
+++ b/cmd/gateway-common.go
@@ -331,6 +331,8 @@ func ErrorRespToObjectError(err error, params ...string) error {
                err = InvalidUploadID{}
        case "EntityTooSmall":
                err = PartTooSmall{}
+       case http.StatusText(http.StatusMethodNotAllowed):
+               err = toObjectErr(errMethodNotAllowed, bucket, object)
        }
 
        return err
```

Reply:
fixed